python/paddle/fluid/variable_index.py

```diff
@@ -0,0 +1,306 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import numpy as np
+from . import unique_name
+from . import core
+
+MAX_INTEGER = 2**31 - 1
```

Comment:
Should we use `sys.maxsize`?

Reply:
No. 
`sys.maxsize is 2**63 - 1`.  Because the maximum of  c++ `int` is `2**31 - 1`,  pass `2**63 - 1` from Python to c++ will result in data out of bounds.